src/Button/ButtonBase.tsx

```diff
@@ -18,6 +19,7 @@ const trailingIconStyles = {
 const ButtonBase = forwardRef<HTMLElement, ButtonProps>(
   ({children, as: Component = 'button', sx: sxProp = defaultSxProp, ...props}, forwardedRef): JSX.Element => {
     const {leadingIcon: LeadingIcon, trailingIcon: TrailingIcon, variant = 'default', size = 'medium', ...rest} = props
+    const innerRef = React.useRef<HTMLElement>()
```

Comment:
```suggestion
    const innerRef = React.useRef<HTMLElement>()
    useRefObjectAsForwardedRef(forwardedRef, innerRef)
```

Here was that convention that I was talking about! This should be available in `src/hooks/useRefObjectAsForwardedRef.ts`. This seems to show up in a couple components that both have a local and forwardedRef

Reply:
AMAZING thank you!